### PR TITLE
Fix Rotation.from_rotvec autodiff at 0

### PR DESCRIPTION
**Pull request by martvald**
Update `from_rotvec` to use `lax.cond` to only evaluate a selected branch. This fixes an issue where differentiating `from_rotvec` return nans in the case of the rotvec being all zeros.

~~I am unsure if I should `from jax._src.lax.control_flow import cond` or use `import jax.lax as lax` and `lax.cond` inside of `_src`. I'll change as adviced.~~

To recreate the issue:
```python
import jax
import jax.lax as lax
import jax.numpy as jnp
from jax import Array
from jax.scipy.spatial.transform import Rotation


def _small_scale(angle2: Array, rotvec: Array) -> Array:
    scale = 0.5 - angle2 / 48.0 + angle2 * angle2 / 3840.0
    w = 1.0 - angle2 / 8.0 + angle2 * angle2 / 384.0
    return jnp.hstack([scale * rotvec, jnp.array([w], rotvec.dtype)])


def _large_scale(angle2: Array, rotvec: Array) -> Array:
    angle = jnp.sqrt(angle2)
    scale = jnp.sin(angle * 0.5) / angle
    xyz = scale * rotvec
    w = jnp.cos(angle * 0.5)
    return jnp.hstack([xyz, jnp.array([w], rotvec.dtype)])


def _from_rotvec(rotvec: Array, degrees: bool) -> Array:
    rotvec = jnp.where(degrees, jnp.deg2rad(rotvec), rotvec)
    angle2 = jnp.dot(rotvec, rotvec)
    return lax.cond(angle2 < 1e-6, _small_scale, _large_scale, angle2, rotvec)


rvec = jnp.array([0, 0, 0], dtype=jnp.float32)

original = jax.jacrev(Rotation.from_rotvec, argnums=0)(rvec, degrees=False).quat
new = jax.jacrev(_from_rotvec, argnums=0)(rvec, degrees=False)

print(original)
print(new)
```
prints:
```
[[nan nan nan]
 [nan nan nan]
 [nan nan nan]
 [nan nan nan]]
[[0.5 0.  0. ]
 [0.  0.5 0. ]
 [0.  0.  0.5]
 [0.  0.  0. ]]
```

### CODE REVIEW

File: jax/_src/scipy/spatial/transform.py [hunk 1]
```diff
@@ -381,15 +382,24 @@ def _elementary_quat_compose(angles: Array, axes: Array, intrinsic: bool, degree
...
 def _from_rotvec(rotvec: Array, degrees: bool) -> Array:
   rotvec = jnp.where(degrees, jnp.deg2rad(rotvec), rotvec)
-  angle = _vector_norm(rotvec)
-  angle2 = angle * angle
-  small_scale = scale = 0.5 - angle2 / 48 + angle2 * angle2 / 3840
-  large_scale = jnp.sin(angle / 2) / angle
-  scale = jnp.where(angle <= 1e-3, small_scale, large_scale)
-  return jnp.hstack([scale * rotvec, jnp.cos(angle / 2)])
+  angle2 = jnp.dot(rotvec, rotvec)
+  return lax.cond(angle2 < 1e-6, _small_scale, _large_scale, angle2, rotvec)
```
Comment (jakevdp) [inline jax/_src/scipy/spatial/transform.py:402]: The first argument to `lax.cond` must be a scalar, and in general I believe `angle2` will be a vector.

**Comment by jakevdp** [comment]
Hey thanks for this. You should be aware that under `vmap`, `cond` becomes a `select`, so if you're trying to fix a `NaN` gradient issue this will not fix the issue in all cases. A more robust fix would be to use the double-where trick, as mentioned at https://docs.jax.dev/en/latest/faq.html#gradients-contain-nan-where-using-where

Also, please note that `jax.scipy.spatial` is not something we are actively working on, and we hope to remove it as soon as scipy itself has a JAX-compatible implementation (see https://docs.jax.dev/en/latest/jep/18137-numpy-scipy-scope.html#scipy-spatial).

**Comment by martvald** [comment]
@jakevdp Thank you, I will close this as there is no reason to merge this if `scipy` is heading out.
